src/core/net/ip6.cpp

```diff
@@ -1068,6 +1072,26 @@ Error Ip6::SendRaw(OwnedPtr<Message> aMessagePtr)
 
     if (header.GetDestination().IsMulticast())
     {
+#if OPENTHREAD_CONFIG_FILTER_LOCAL_ORIGINATED_MULTICAST_PACKETS
```

Comment:
Should we add a new API to control the behavior at runtime? I think this is a security feature, and should be controlled by upper layer.

Reply:
I'll update this PR to add API control for it.